src/lib.rs

```diff
@@ -34,6 +34,32 @@ pub enum Error {
     OutOfMemory,
 }
 
+impl std::error::Error for crate::Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match *self {
+            Error::EOF => None,
+            Error::BadInput => None,
+            Error::UnknownColorspace => None,
+            Error::ParseError(_) => None,
+            Error::IoError(_) => None,
```

Comment:
In these two cases it seems to me that there is an underlying error.

Reply:
I wasn't sure how to tackle these... Updated :)